### PR TITLE
Reduce confusion in testing documentation.

### PR DESCRIPTION
**Pull request by cpojer**
Just wanted to add some clarity to this page and link to the react tutorial on Jest's website to give people more information. Also changed enzyme's definition from library to utility to help reduce the confusion. Hope this makes sense to everybody :)

See https://twitter.com/damusnet/status/780752042675597312

cc @lelandrichardson

### CODE REVIEW

File: docs/docs/10.4-test-utils.md [hunk 1]
```diff
@@ -6,15 +6,15 @@ prev: two-way-binding-helpers.html
...
+`ReactTestUtils` makes it easy to test React components in the testing framework of your choice. At Facebook we use [Jest](https://facebook.github.io/jest/) for painless JavaScript testing. Learn how to get started with Jest through the Jest website's [React Tutorial](http://facebook.github.io/jest/docs/tutorial-react.html#content).
 
 ```
 var ReactTestUtils = require('react-addons-test-utils');
 ```
 
 > Note:
 > 
-> Airbnb has released a testing utility called Enzyme, which makes it easy to assert, manipulate, and traverse your React Components' output. If you're deciding on a unit testing library, it's worth checking out: [http://airbnb.io/enzyme/](http://airbnb.io/enzyme/)
+> Airbnb has released a testing utility called Enzyme, which makes it easy to assert, manipulate, and traverse your React Components' output. If you're deciding on a unit testing utility to use together with Jest, it's worth checking out: [http://airbnb.io/enzyme/](http://airbnb.io/enzyme/)
```
Comment (Aweary) [inline docs/docs/10.4-test-utils.md]: I fear that people might interpret this to mean that Enzyme is a utility specifically for Jest

Comment (cpojer) [inline docs/docs/10.4-test-utils.md]: All I care about is the distinction that enzyme != Jest. Do you have a suggestion on how to improve the sentence? :)

Comment (Aweary) [inline docs/docs/10.4-test-utils.md]: I'm not sure, maybe add something to make it clear that it enzyme can be used with other testing libraries. Something like:

> If you're deciding on a unit testing utility to use together with Jest, or another testing library, it's worth checking out

**Comment by cpojer** [comment]
Updated to mention other "test runners". Sounds good?

**Comment by Aweary** [comment]
@cpojer sounds good to me! 🎉

**Comment by vjeux** [comment]
It's going live as soon as travis pushes it :)

**Comment by vjeux** [comment]
One day maybe... https://travis-ci.org/facebook/react/builds/163285316

**Comment by gaearon** [comment]
So that’s where the confusion comes from! Thanks for spotting.

**Comment by lelandrichardson** [comment]
👍

**Pull request merged**